frontend/src/components/GitHubLoginDialog.tsx

```diff
@@ -60,7 +61,10 @@ export function GitHubLoginDialog({
           setPolling(false);
           setDeviceState(null);
           setError(null);
-          window.location.reload(); // reload config
+          // Add a small delay to ensure backend has time to save the token
+          setTimeout(() => {
+            onOpenChange(false);
+          }, 500);
```

Comment:
we don't necessarily need this delay as we already have one inside handleGitHubLoginComplete